indexer-app/test/cmr/indexer/test/services/index_set_service_test.clj

```diff
@@ -225,58 +225,178 @@
 
 (deftest test-get-concept-type-for-index
-  (testing "returns concept type when index found for granule"
+  (testing "returns concept type for regular index without shards"
     (let [index-set {:index-set
-                     {:concepts {:granule {"granule-v1" "1_cmr_granules_v1"
-                                           "granule-v2" "1_cmr_granules_v2"}
-                                 :collection {"collection-v1" "1_cmr_collections_v1"}}}}]
+                     {:concepts {:granule {:small_collections "1_small_collections"}}}}]
       (is (= :granule
-             (svc/get-concept-type-for-index index-set "1_cmr_granules_v1")))))
+             (svc/get-concept-type-for-index index-set "1_small_collections")))))
 
-  (testing "returns concept type when index found for collection"
+  (testing "returns concept type for regular index with shards"
     (let [index-set {:index-set
-                     {:concepts {:granule {"granule-v1" "1_cmr_granules_v1"}
-                                 :collection {"collection-v1" "1_cmr_collections_v1"
-                                              "collection-v2" "1_cmr_collections_v2"}}}}]
+                     {:concepts {:granule {:small_collections "1_small_collections_100_shards"}}}}]
+      (is (= :granule
+             (svc/get-concept-type-for-index index-set "1_small_collections_100_shards")))))
+
+  (testing "returns concept type for concept ID style index"
+    (let [index-set {:index-set
+                     {:concepts {:granule {(keyword "C2317033465-NSIDC_ECS") "1_c2317033465_nsidc_ecs"}}}}]
+      (is (= :granule
+             (svc/get-concept-type-for-index index-set "1_c2317033465_nsidc_ecs")))))
+
+  (testing "returns concept type for concept ID with shards"
+    (let [index-set {:index-set
+                     {:concepts {:granule {(keyword "C2317033465-NSIDC_ECS") "1_c2317033465_nsidc_ecs_5_shards"}}}}]
+      (is (= :granule
+             (svc/get-concept-type-for-index index-set "1_c2317033465_nsidc_ecs_5_shards")))))
+
+  (testing "returns concept type for collection index with version"
+    (let [index-set {:index-set
+                     {:concepts {:collection {(keyword "collections-v2") "1_collections_v2"}}}}]
       (is (= :collection
-             (svc/get-concept-type-for-index index-set "1_cmr_collections_v2")))))
+             (svc/get-concept-type-for-index index-set "1_collections_v2")))))
 
-  (testing "returns nil when index not found in any concept type"
+  (testing "returns concept type for variable concept ID"
     (let [index-set {:index-set
-                     {:concepts {:granule {"granule-v1" "1_cmr_granules_v1"}
-                                 :collection {"collection-v1" "1_cmr_collections_v1"}}}}]
-      (is (nil? (svc/get-concept-type-for-index index-set "nonexistent_index")))))
+                     {:concepts {:variable {(keyword "V123-5W5_NSIDC_ECS") "1_v123_5w5_nsidc_ecs"}}}}]
```

Comment:
V123-5W5 isn't a valid concept id.  As code rabbit pointed out the get-canonical-key-name function does not correctly work with V123-5W5_NSIDC_ECS as the concept name. The real value should be 
V123_5W5-NSIDC_ECS, as NSIDC_ECS is the real platform and not 5W5_NSIDC_ECS.  Unless you wanted 5W5_NSIDC_ECS to be the provider name, but something still seems off.